incubator/crate/templates/statefulset.yaml

```diff
@@ -0,0 +1,115 @@
+  replicas: {{ .Values.crate.replicas }}
+  selector:
+    matchLabels:
+      app: {{ .Values.app }}
+  template:
+    metadata:
+      labels:
+        app: {{ .Values.app }}
+        app.kubernetes.io/name: crate-deployment
+        app.kubernetes.io/part-of: crate-db
+        app.kubernetes.io/version: {{ .Values.image.tag }}
+        app.kubernetes.io/managed-by: helm
+      annotations:
+        pod.alpha.kubernetes.io/initialized: "true"
+    spec:
+      initContainers:
+      - name: init-sysctl
+        image: busybox
+        imagePullPolicy: IfNotPresent
+        command: ["sysctl", "-w", "vm.max_map_count=262144"]
+        securityContext:
+          privileged: true
+      containers:
+      - name: crate
+        image: crate:{{ .Values.image.tag }}
+        imagePullPolicy: IfNotPresent
+        command:
+          - /docker-entrypoint.sh
+          - -Clicense.enterprise=false
+          - -Ccluster.initial_master_nodes={{ range $i := until (int .Values.crate.numberOfNodes) -}}
+              {{ if $i }},{{ end }}crate-{{ $i }}
+         {{- end }}
+          - -Ccluster.name={{ .Values.crate.clusterName }}
+          - -Cnetwork.host="0.0.0.0"
+          - -Cnode.name="$POD_NAME"
+          - -Cdiscover.seed_providers=srv
+          - -Cdiscovery.srv.query=_{{ .Values.service.clusterName }}._tcp.{{ .Values.service.name }}.{{ .Release.Namespace }}.svc.cluster.local
+          - -Cgateway.recover_after_nodes={{ if $r := .Values.crate.recoverAfterNodes -}}
+              {{ $r }}{{ else }}{{ div .Values.crate.numberOfNodes 2 | floor | add 1 }}
+         {{- end }}
+          - -Cgateway.expected_nodes={{ .Values.crate.numberOfNodes }}
+          - -Chttp.cors.enabled={{ .Values.http.cors.enabled }}
+          - -Chttp.cors.allow-origin={{ .Values.http.cors.allowOrigin }}
+          - -Cpath.data=/data
+          {{- if .Values.resources.limits.cpu }}
+          - -Cprocessors="$MIN_VCPU_COUNT"
+          {{- end }}
+        volumeMounts:
+            - mountPath: /data
+              name: data
+        resources:
+          requests:
+            cpu: {{ .Values.resources.requests.cpu }}
+            memory: {{ mul .Values.crate.heapSize 2 }}Gi
+          limits:
+            cpu: {{ .Values.resources.limits.cpu }}
+            memory: {{ if $m := .Values.resources.limits.memory -}}
+              {{ $m }}{{ else }}{{ mul .Values.crate.heapSize 3 }}Gi
+            {{- end }}
+        ports:
+        - containerPort: {{ .Values.service.ports.psql }}
+          name: psql
+        - containerPort: 4300
+          name: transport
+        - containerPort: {{ .Values.service.ports.ui }}
+          name: ui
+        env:
+        - name: CRATE_HEAP_SIZE
+          value: {{ .Values.crate.heapSize }}g
+        - name: CLUSTER_NAME
+          value: {{ .Values.crate.clusterName | quote }}
+        - name: MIN_VCPU_COUNT
+          valueFrom:
+            resourceFieldRef:
+              resource: requests.cpu
+              divisor: 1
```

Comment:
This might still be fractional, I'm struggling to parse the docs, need to verify.

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#quantity-resource-core

Reply:
Nope, this works great. I tested on a deployment with `requests/cpu: 128m`, and this `resourceFieldRef` evaluates to `1`, which is exactly what we want for Crate's `-Cprocessors` flag.